docker-compose/node-entrypoint.sh

```diff
@@ -2,19 +2,22 @@
 
 set -euxo pipefail
 
+genesis_data_path="${GENESIS_DATA_PATH-}"
+data_dir="${CONCORDIUM_NODE_DATA_DIR}"
```

Comment:
Why do this renaming? CONCORDIUM_NODE_DATA_DIR is the environment variable the node uses, it seems to introduce needless overhead to rename a variable.

Reply:
The style that we've found useful in bash in general is to extract all expected variables up front with the convention to only use them as lower-case variables afterwards (to separate internal from external variables and avoid leaking). This improves error handling by ensuring that the script doesn't crash mid-way because something is missing (the `set -u` flag makes it crash immediately) and that default values are processed in a single location. It also makes it easier to see from the source which variables are expected.

Reply:
This is contradicted by the use of BAKER_ID_URL below.

I find this style annoying since it makes it harder to understand what is going on or trace configurations since there is always many different names for the thing you are looking for so searching is hard.

Reply:
Yeah that's because `BAKER_ID_URL` is required only if `GENESIS_DATA_PATH` is set. I found that adding the logic to handle that up front was overkill. So no the convention is not perfect but I've found it to simplify debugging tremendously time and again. Not that I expect it to matter much with a script this simple, I just like to be consistent about it.

Reply:
It would alleviate the annoyance I have greatly if the convention was to just lowercase the variables instead of giving them different names since searching with "ignore case" is well supported in tools.

Reply:
done